src/components/FileInput.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from "react";
+import "../css/FileInput.css";
+
+const FileInput = ({ value, onChange, onInputChange }) => {
+  const { tags, price, description, name } = value;
+  const [preview, setPreview] = useState(null);
+  const [imgError, setImgError] = useState("");
+
+  const handleChange = (e) => {
+    // 이미지 세트함수
+    const nextValue = e.target.files[0];
+    if (preview) {
+      setImgError("*이미지 등록은 최대 1개까지 가능합니다.");
+      return;
+    }
+    onChange("images", nextValue);
+
+    if (nextValue) {
+      // 이미지 프리뷰
+      const imgURL = URL.createObjectURL(nextValue);
+      setPreview(imgURL);
+      return () => {
+        setPreview();
+        URL.revokeObjectURL(nextValue);
+      };
+    }
+  };
+
+  const handleRemovePreview = (preview) => {
+    //이미지 프리뷰, 에러메세지 삭제
+    setPreview(null);
+    setImgError("");
+  };
+
+  const handleTagChange = (e) => {
+    //태그 입력 및 엔터
+    const newTag = e.target.value.trim();
+    if (e.key === "Enter" && newTag !== "") {
+      e.preventDefault();
+      const updatedTags = [...tags, newTag];
+      onChange("tags", updatedTags);
+      e.target.value = "";
+    }
+  };
+
+  const handleRemoveTag = (index) => {
+    // 태그 제거
+    const updatedTags = tags.filter((_, i) => i !== index);
+    onChange("tags", updatedTags);
+  };
+
+  const isFormValid = () => {
```

Comment:
함수 리턴값으로 받아도 상관은 없지만 아래와 같이 간결하게 작성할 수도 있습니다 :)
```suggestion
  const isFormValid = name && description && price && !!tags.length
```